app/src/main/java/com/nervousfish/nervousfish/activities/SettingsActivity.java

```diff
@@ -18,17 +18,23 @@
 import com.nervousfish.nervousfish.R;
 import com.nervousfish.nervousfish.data_objects.Profile;
 import com.nervousfish.nervousfish.service_locator.IServiceLocator;
+import com.nervousfish.nervousfish.service_locator.NervousFish;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.util.List;
 
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import edu.umd.cs.findbugs.annotations.SuppressWarnings;
+
 /**
  * A {@link PreferenceActivity} that presents a set of application settings. This
  * is the place where your profile is changed.
  */
+@SuppressFBWarnings(value = "ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
+//Because of the structure with the static methods, a static attribute has to be set.
```

Comment:
This is the default usage of the settings screen. The only place where the static method is important is at the on preference change listener. This is one method which is triggered for all changes in preferences from the settings screen. Because this method uses the servicelocator, we have to set that in the onCreate.

I'm actually thinking that maybe we can statically get it from the NervousFish class?